### ISSUE TITLE
Can't load primary/alignment files only

### ISSUE THREAD

**Issue by ces** [wtsi-npg/npg_irods#207]
If we just want to load primary files e.g. cram, crai, flagstat and not qc files - as after ad-hoc reprocessing - the npg_publish_illumina_run.pl script currently fails as it relies on the bam_flagstats.json file to find the read count for the cram meta data. 

e.g. npg_publish_illumina_run.pl --driver-type samplesheet --id_run=$run --include=lane$lane --exclude=lane$lane/qc --verbose --source-directory=.

2019/01/28 16:59:01 ERROR WTSI.NPG.HTS.Illumina.RunPublisher - Unexpected error in publish_files: Found 0 matching '(?^msx:\/6683_4\#13\.bam_flagstats\.json$)' where one was expected: [] at /blah/lib/perl5/WTSI/NPG/HTS/Illumina/RunPublisher.pm line 696. 

Be good to have a option or way to exclude qc files from loading as previously.

**Comment by keithj** [comment]
We'll have to consider this carefully.  This new behaviour arises because given this will be expected to work on arbitrary file hierarchies, so the loader respecting the include/exclude lists is key to giving enough control over what files are processed. However, it also means that if you exclude some files, they really are excluded, even when the included files might need them for metadata. You should be able to supply an exclude pattern that matches the QC files, except `bam_flagstats` e.g. `--exclude='plex1/qc/(?!bam_flagstats.json)$'` but then you will get the `bam_flagstats` files published.

**Comment by kjsanger** [comment]
Yes, this is a result of the exclusion pattern excluding a file that is required to load one of the included files. To do this we'd have to add the concept of a filtered resultset (containing just the files you want to load) which is a subset of the full resultset (which contains other files, you don't want to load, but may want to operate on).
